transaction/simple.go

```diff
@@ -2,29 +2,35 @@ package transaction
 
 import (
 	"github.com/gluster/glusterd2/context"
+
+	"github.com/pborman/uuid"
 )
 
 // SimpleTxn is transaction with fixed stage, commit and store steps
 type SimpleTxn struct {
```

Comment:
Taking an operation such as volume create as example, I'd prefer to see some detailed comment as to how this mechanism will adhere (resilient to TOCTOU and network disruptions) to some well known protocols such as 2PC or 3PC/E3PC which has been widely implemented and has formal proofs. If I missed this in design docs or specs, please point me there.